server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/MapStringTermsAggregator.java

```diff
@@ -383,6 +386,9 @@ void collectZeroDocEntriesIfNeeded(long owningBucketOrd) throws IOException {
                 SortedBinaryDocValues values = valuesSource.bytesValues(ctx);
                 // brute force
                 for (int docId = 0; docId < ctx.reader().maxDoc(); ++docId) {
+                    if (excludeDeletedDocs && ctx.reader().getLiveDocs() != null && ctx.reader().getLiveDocs().get(docId) == false) {
```

Comment:
You may want to extract the result of `ctx.reader().getLiveDocs()` since `getLiveDocs()` performs a call to `ensureOpen()` under the hood which is not completely free.